grammars/css.cson

```diff
@@ -20,294 +20,1205 @@
+          (?![\\w-])
+        '''
+        'name': 'support.constant.color.w3c-extended-color-name.css'
+      }
+      {
+        # Current text colour
+        'match': '(?i)(?<![\\w-])currentColor(?![\\w-])'
+        'name': 'support.constant.color.current.css'
+      }
+      {
+        # These colours are deprecated in CSS3: http://www.w3.org/TR/css3-color/#css2-system
+        'match': '''(?xi) (?<![\\w-])
+          (ActiveBorder|ActiveCaption|AppWorkspace|Background|ButtonFace|ButtonHighlight|ButtonShadow
+          |ButtonText|CaptionText|GrayText|Highlight|HighlightText|InactiveBorder|InactiveCaption
+          |InactiveCaptionText|InfoBackground|InfoText|Menu|MenuText|Scrollbar|ThreeDDarkShadow
+          |ThreeDFace|ThreeDHighlight|ThreeDLightShadow|ThreeDShadow|Window|WindowFrame|WindowText)
+          (?![\\w-])
+        '''
+        'name': 'invalid.deprecated.color.system.css'
+      }
+    ]
+  'combinators':
+    'patterns': [
+      {
+        'match': '/deep/|>>>'
+        'name': 'invalid.deprecated.combinator.css'
+      }
+      {
+        'match': '>|\\+|~|>>'
+        'name': 'keyword.operator.combinator.css'
+      }
+    ]
+  'commas':
+    'match': ','
+    'name': 'punctuation.separator.list.comma.css'
+  'comment-block':
+    'begin': '/\\*'
+    'beginCaptures':
       '0':
-        'name': 'keyword.control.at-rule.charset.css'
-      '1':
-        'name': 'punctuation.definition.keyword.css'
-    'end': '\\s*((?=;|$))'
-    'name': 'meta.at-rule.charset.css'
+        'name': 'punctuation.definition.comment.begin.css'
+    'end': '\\*/'
+    'endCaptures':
+      '0':
+        'name': 'punctuation.definition.comment.end.css'
+    'name': 'comment.block.css'
+  'escapes':
     'patterns': [
       {
-        'include': '#string-double'
+        'match': '\\\\[0-9a-fA-F]{1,6}'
+        'name': 'constant.character.escape.codepoint.css'
       }
       {
-        'include': '#string-single'
+        'match': '\\\\$\\n?'
+        'name': 'constant.character.escape.newline.css'
       }
       {
-        'include': '#comment-block'
+        'match': '\\\\.'
+        'name': 'constant.character.escape.css'
       }
     ]
-  }
-  {
-    'begin': '(@)import\\b'
-    'captures':
+  'feature-query':
+    'begin': '\\('
+    'beginCaptures':
+      '0':
+        'name': 'punctuation.definition.condition.begin.bracket.round.css'
+    'end': '\\)'
+    'endCaptures':
       '0':
-        'name': 'keyword.control.at-rule.import.css'
+        'name': 'punctuation.definition.condition.end.bracket.round.css'
+    'name': 'meta.feature-query.css'
+    'patterns': [
+      {
+        'include': '#feature-query-operators'
+      }
+      {
+        'include': '#feature-query'
+      }
+    ]
+  'feature-query-operators':
+    'patterns': [
+      {
+        'match': '(?i)(?<=[\\s()]|^|\\*/)(and|not|or)(?=[\\s()]|$|/\\*)'
+        'name': 'keyword.operator.logical.$1.feature.css'
```

Comment:
So, this now becomes:

```
keyword.operator.logical.feature.and.css
keyword.operator.logical.feature.not.css
keyword.operator.logical.feature.or.css
```

Which looks very silly, but I assume you have your reasons for this. =)